### PR TITLE
layout: Separate out overflow-for-scrolling from overflow-for-paint.

### LINKED ISSUE #9484: Unexpected overflow
STR:
1) place mouse over the green div
2) scroll horizontally and/or vertically

The page will start moving, even though the parent div is `100%` and `overflow:hidden`.

Scrolling doesn't alway happen. Repeat step (2).

The  behavior is not exactly the same with Webrender. But both regular Servo and Webrender have this overflow.

```html
<style>
  body {
    margin: 0;
  }

  .red {
    background: red;
    position: absolute;
    width: 100%;
    height: 100%;
    overflow: hidden;
  }

  .green {
    background: green;
    position: absolute;
    width: 500px;
    height: 100%;
    right: 0px;

    width: 500px;
    transform: translateX(250px);
    box-shadow: 0px 0px 200px black;
  }
</style>

<div class="red">
  <div class="green"></div>
</div>
```

### PR DESCRIPTION
**Pull request by pcwalton**
Closes #9484.

r? @mbrubeck

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9522)

### CODE REVIEW

File: tests/html/overflow-box-shadow.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<!-- Nothing on this page should be scrollable. ->
```
Comment (timvandermeij) [inline tests/html/overflow-box-shadow.html]: Does `->` need to be `-->` for the comment to work?

**Comment by mbrubeck** [comment]
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 16 of 17 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[components/layout/fragment.rs, line 2203 \[r2\]](https://reviewable.io:443/reviews/servo/servo/9522#-K9dMDlXzyd7awNiYHVn:-K9dMDlYa1SlVftPiFn9:856649758) ([raw file](https://github.com/servo/servo/blob/797ccb2d673275bd9cd58e16cbdcdbd8476e5f1b/components/layout/fragment.rs#L2203)):*
I think outlines should only affect the paint overflow.  “The outline created with the outline properties is drawn "over" a box, i.e., the outline is always on top, and does not influence the position or size of the box, or of any other boxes.”

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9522#-:-K9dMYt0dZVaEgzOHbyf:1458266033)*

**Comment by mbrubeck** [comment]
Can land with r=mbrubeck with the two review comments addressed.

**Comment by KiChjang** [comment]
@bors-servo retry
* #9148

**Comment by jdm** [comment]
@bors-servo: retry
* #9033

**Comment by Wafflespeanut** [comment]
@bors-servo retry

#9033

**Pull request merged**